Use Sharp if available for Image Optimization (#27346)
vercel/next.js: docs/basic-features/image-optimization.md

The following Image Optimization cloud providers are included:

- [Vercel](https://vercel.com): Works automatically when you deploy on Vercel, no configuration necessary. [Learn more](/docs/next.js/image-optimization)
- [Imgix](https://www.imgix.com): `loader: 'imgix'`
- [Cloudinary](https://cloudinary.com): `loader: 'cloudinary'`
- [Akamai](https://www.akamai.com): `loader: 'akamai'`

> - Custom: `loader: 'custom'` use a custom cloud provider by implementing the [`loader`](https://vercel.com/docs/api-reference/next/image.md#loader) prop on the `next/image` component

Reviewer: This needs to remain `/docs` because its nextjs.org

The following Image Optimization cloud providers are included:

> - [Vercel](https://vercel.com): Works automatically when you deploy on Vercel, no configuration necessary. [Learn more](/docs/next.js/image-optimization)

Reviewer: This needs to remain vercel.com because the relative path will be incorrect without it (remember this is deployed to nextjs.org)

Author: Ok, that makes sense. I was wondering why this one didn't match up with the others on the page

> The `next/image` component's default loader is not supported when using [`next export`](/docs/advanced-features/static-html-export.md). However, other loader options will work.

> > By default, Next.js uses the ['squoosh'](https://github.com/GoogleChromeLabs/squoosh/tree/dev/cli) library for image resizing and optimization. This library is quick to install and suitable for a dev server environment. If you are going to use the default image loader and built-in image optimizer in production, it is strongly recommended that you install the optional [`sharp`](https://github.com/lovell/sharp) image optimization library. Do this by running `yarn add sharp` in your application directory.

Reviewer: ```suggestion
> The `next/image` component's default loader uses the ['squoosh'](https://www.npmjs.com/package/@squoosh/lib) library for image resizing and optimization. This library is quick to install and suitable for a dev server environment. For a production environment, it is strongly recommended that you install the optional [`sharp`](https://www.npmjs.com/package/sharp) library by running `yarn add sharp` in your project directory.
```
